src/sysinfo/nvapi_adapter.cpp

```diff
@@ -200,7 +204,7 @@ namespace dxvk {
     }
 
     NV_GPU_ARCHITECTURE_ID NvapiAdapter::GetArchitectureId() const {
-        if (!this->HasNvProprietaryDriver()) {
+        if (!this->HasNvProprietaryDriver() && !this->HasNvkDriver()) {
```

Comment:
This is one place I'm not too sure about. Except for Ada GPUs, detection based on support of various Vulkan extensions might be… not so reliable until NVK gets support for them, and there's no guarantee NVK will bother implementing NV(X) vendor extensions.

An idea for another day but maybe we could swap the extensions checked for something less specific to whatever NV proprietary driver supports on each arch? E.g. check for `VK_EXT_mesh_shader` to detect Turing-or-newer?

Reply:
also, correct me if I'm wrong, but should it not be an || instead of && ?

Reply:
> This is one place I'm not too sure about. Except for Ada GPUs, detection based on support of various Vulkan extensions might be… not so reliable until NVK gets support for them, and there's no guarantee NVK will bother implementing NV(X) vendor extensions.
> 
> An idea for another day but maybe we could swap the extensions checked for something less specific to whatever NV proprietary driver supports on each arch? E.g. check for `VK_EXT_mesh_shader` to detect Turing-or-newer?

ah yeah, thanks, that's a very valid point. Using `VK_EXT_mesh_shader` sounds indeed good. I would also prefer to have a valid detection for the older generations. From what I remember, NVK is currently Pascal and newer so we would for now just need something to identify `Volta`. That would change ofc once NVK starts to support even older generations. I don't know if there is a limit for NVK.

@Sid127 the check should be good, note the `!` , also the test seem to confirm that it works correctly.

Reply:
PS: https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/nouveau/vulkan/nvk_physical_device.c?ref_type=heads#L230 and https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/27196